CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+cmake_minimum_required (VERSION 2.6)
+project (LIBUAST C CXX)
+
+# C++ 11 required
+set (CMAKE_CXX_STANDARD 11)
+
+# Required dependencies
+find_package(LibXml2 REQUIRED)
+
+# Source files
+set(libsrc node.cpp roles.cpp)
+
+# Header files
+set(libinclude node.h roles.h)
+
+# Include header
+include_directories(
+  ${LIBXML2_INCLUDE_DIR}
+  ${CMAKE_BINARY_DIR})
+
+# Generate libuast.a static library
+add_library(uast STATIC ${libsrc})
+
+set_target_properties(uast PROPERTIES PUBLIC_HEADER "${libinclude}")
+
+# Install static library
+install(
+  TARGETS       uast
+  ARCHIVE       DESTINATION lib
+  LIBRARY       DESTINATION lib
+  PUBLIC_HEADER DESTINATION include/libuast
+)
+
+############################################
+# Generate executable to test static lib
+############################################
+
+add_executable(example1.out test.cpp)
```

Comment:
I'd separate the example build rules since they're not really needed to build the library and it can serve as an example of how a C/C++ user would use the library. An `examples` directory containing both a C and a C++ example would be great. It'd be nice to have some examples using C++ generated protobuf interacting with the library, but that may be left for the future.